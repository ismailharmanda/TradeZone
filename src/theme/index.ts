export const theme = {
  radius: 8,
  text: {
    fontFamily: 'Silka, sans-serif',
    size: {
      xl: {
        fontSize: 28,
        lineHeight: 33.6,
      },
      lg: {
        fontSize: 20,
        lineHeight: 21,
      },
      md: {
        fontSize: 16,
        lineHeight: 21,
      },
      sm: {
        fontSize: 14,
        lineHeight: 16.8,
      },
      xs: {
        fontSize: 12,
        lineHeight: 14.4,
      },
    },
  },
  colors: {
    base: {
      transparent: 'rgba(255, 255, 255, 0)',
      black: '#000000',
      white: '#ffffff',
      dark: '#5A5A5A',
    },
    primary: {
      regular: 'rgb(255,119,5)',
      transparent: 'rgba(255,119,5,0.6)',
    },
    secondary: {
      regular: 'rgb(240,237,246)',
      transparent: 'rgba(240,237,246,0.6)',
    },
    tertiary: {
      regular: 'rgb(0,110,94)',
      transparent: 'rgba(0,110,94,0.6)',
    },
    success: {
      regular: '#006E5E',
    },
    error: {
      regular: '#FF0000',
    },
  },
  shadow: {
    elevation1: {
      shadowColor: 'rgba(0, 0, 0, 1)',
      shadowOffset: {
        width: 3,
        height: 3,
      },
      shadowOpacity: 0.1,
      shadowRadius: 10,
    },
    elevation2: {
      shadowColor: 'rgba(0, 0, 0, 1)',
      shadowOffset: {
        width: 3,
        height: 3,
      },
      shadowOpacity: 0.3,
      shadowRadius: 10,
    },
    elevation3: {
      shadowColor: 'rgba(0, 0, 0, 0.05)',
      shadowOffset: {
        width: 0,
        height: 4,
        blur: 4,
      },
      shadowRadius: 7,
    },
  },
  spacing: {
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 48,
    screenHorizontalPadding: 16,
    screenVerticalPadding: 24,
  },
  height: {
    button: 48,
    input: 48,
  },
};
