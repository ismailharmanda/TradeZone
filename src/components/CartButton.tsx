import {TouchableOpacity} from 'react-native';
import {theme} from 'theme';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {useNavigation} from '@react-navigation/native';

export const CartButton = () => {
  const navigation = useNavigation();

  const onCartPress = () => {
    navigation.navigate('CartStack');
  };

  return (
    <TouchableOpacity onPress={onCartPress}>
      <MaterialCommunityIcons
        name="cart"
        size={26}
        color={theme.colors.primary.regular}
      />
    </TouchableOpacity>
  );
};
