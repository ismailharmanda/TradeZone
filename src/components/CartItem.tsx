import {
  Text,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
  Image,
  View,
} from 'react-native';
import {theme} from 'theme';
import {CartProduct} from 'screens/Cart/cart';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {toFixedNumber} from 'utils';

interface Props {
  product: CartProduct;
  onPlusPress: (product: CartProduct) => void;
  onMinusPress: (product: CartProduct) => void;
  onDelete: (product: CartProduct) => void;
}

const HEIGHT = Dimensions.get('window').height;

export const CartItem = ({
  onPlusPress,
  onMinusPress,
  onDelete,
  product,
}: Props) => {
  return (
    <View style={styles.container}>
      <Image
        style={styles.image}
        resizeMode="contain"
        source={{
          uri: product.image,
        }}
      />
      <View style={styles.textWrapper}>
        <Text numberOfLines={3} style={styles.textTitle}>
          {product.title}
        </Text>
        <Text numberOfLines={1} adjustsFontSizeToFit style={styles.textPrice}>
          {product.price} Ã— {product.quantity} ={' '}
          {toFixedNumber(product.price * product.quantity, 2)} TL
        </Text>
      </View>
      <View style={styles.calculatorWrapper}>
        <View style={styles.calculator}>
          <TouchableOpacity onPress={() => onMinusPress(product)}>
            <Text style={styles.calculatorButton}>-</Text>
          </TouchableOpacity>
          <Text style={styles.textQuantity}>{product.quantity}</Text>
          <TouchableOpacity onPress={() => onPlusPress(product)}>
            <Text style={styles.calculatorButton}>+</Text>
          </TouchableOpacity>
        </View>
        <TouchableOpacity onPress={() => onDelete(product)}>
          <MaterialCommunityIcons
            name="trash-can-outline"
            color={theme.colors.base.dark}
            size={26}
          />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderRadius: theme.radius,
    overflow: 'hidden',
    backgroundColor: theme.colors.secondary.transparent,
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    height: HEIGHT / 7,
    borderWidth: 1,
    borderColor: theme.colors.secondary.regular,
    paddingHorizontal: theme.spacing.sm,
    gap: theme.spacing.sm,
    paddingVertical: theme.spacing.sm,
  },
  calculator: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  calculatorWrapper: {
    height: '100%',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {width: '20%', height: '100%', flex: 1},
  button: {
    width: '100%',
    height: theme.height.button,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: theme.radius,
  },
  textTitle: {
    fontWeight: 'bold',
    fontSize: theme.text.size.xs.fontSize,
    lineHeight: theme.text.size.xs.lineHeight,
    color: theme.colors.base.dark,
    width: '100%',
    flexWrap: 'wrap',
  },

  textPrice: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.primary.regular,
    width: '100%',
    flexWrap: 'wrap',
  },
  textQuantity: {
    fontSize: theme.text.size.lg.fontSize,
  },
  textWrapper: {
    justifyContent: 'space-evenly',
    height: '100%',
    flex: 2,
  },
  calculatorButton: {
    fontSize: theme.text.size.lg.fontSize,
    lineHeight: theme.text.size.lg.lineHeight,
    fontWeight: 'bold',
    color: theme.colors.primary.regular,
    padding: theme.spacing.md,
  },
});
