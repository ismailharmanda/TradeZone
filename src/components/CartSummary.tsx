import {Text, StyleSheet, View} from 'react-native';
import {theme} from 'theme';
import {toFixedNumber} from 'utils';

interface Props {
  subTotal: number;
  discounts: {
    [key: string]: {
      total: number;
      summary: number;
      discountAmount: number;
      percentage: number;
    };
  };
  summary: number;
}

export const CartSummary = ({subTotal, discounts, summary}: Props) => {
  const categories: string[] = Object.keys(discounts);
  return (
    <View style={styles.container}>
      <Text style={styles.textTitle}>Sub Total: {subTotal} TL</Text>
      <View>
        {!!categories.length && (
          <Text style={styles.textDiscountTitle}>Discounts:</Text>
        )}
        {categories.map(category => {
          if (discounts[category].discountAmount === 0) {
            return null;
          }
          return (
            <View key={category}>
              <Text style={styles.textCategory}>
                <Text style={styles.textCategoryTitle}>{category}</Text>{' '}
                Category{' '}
                <Text style={styles.textCategoryPercentage}>
                  {discounts[category].percentage * 100}%
                </Text>{' '}
                - {discounts[category].discountAmount} TL
              </Text>
            </View>
          );
        })}
      </View>
      <Text style={styles.textSummary}>
        Summary:{' '}
        <Text style={styles.textSummaryPrice}>{toFixedNumber(summary, 2)}</Text>{' '}
        TL
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.secondary.transparent,
    padding: theme.spacing.sm,
    borderRadius: theme.radius,
    borderWidth: 1,
    borderColor: theme.colors.tertiary.transparent,
    width: '100%',
    gap: theme.spacing.md,
  },
  textTitle: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.base.dark,
    width: '100%',
    flexWrap: 'wrap',
  },
  textCategoryTitle: {
    color: theme.colors.tertiary.regular,
    textTransform: 'capitalize',
  },
  textCategory: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.base.dark,
  },
  textCategoryPercentage: {
    color: theme.colors.primary.regular,
  },
  textSummary: {
    fontWeight: 'bold',
    fontSize: theme.text.size.lg.fontSize,
    lineHeight: theme.text.size.lg.lineHeight,
    color: theme.colors.base.dark,
  },
  textSummaryPrice: {
    color: theme.colors.primary.regular,
  },
  textDiscountTitle: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.primary.regular,
    marginBottom: theme.spacing.xs,
  },
});
