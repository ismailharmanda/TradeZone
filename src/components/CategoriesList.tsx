import {useCallback} from 'react';
import {
  FlatList,
  ViewStyle,
  ListRenderItem,
  View,
  StyleSheet,
} from 'react-native';
import {Categories} from 'screens/Products/products';
import {theme} from 'theme';

interface Props {
  data: Categories;
  renderItem: ListRenderItem<string>;
  ListEmptyComponent: JSX.Element;
  contentContainerStyle?: ViewStyle;
  listStyle?: ViewStyle;
}

export const CategoriesList = ({
  data,
  renderItem,
  ListEmptyComponent,
  contentContainerStyle,
  listStyle,
}: Props) => {
  const keyExtractor = useCallback((item: any) => item, []);

  const ItemSeperator = useCallback(
    () => <View style={styles.itemSeperator} />,
    [],
  );

  return (
    <View testID="categoriesList" style={styles.container}>
      <FlatList
        testID="flatList"
        showsHorizontalScrollIndicator={false}
        ItemSeparatorComponent={ItemSeperator}
        horizontal
        style={[listStyle]}
        contentContainerStyle={contentContainerStyle}
        keyExtractor={keyExtractor}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: theme.spacing.xl,
  },
  itemSeperator: {
    width: theme.spacing.md,
  },
});
