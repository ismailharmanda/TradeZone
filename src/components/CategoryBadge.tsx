import {Text, StyleSheet, TouchableWithoutFeedback, View} from 'react-native';
import {theme} from 'theme';

interface Props {
  onPress: (item: string) => void;
  activeCategory: string;
  item: string;
}

export const CategoryBadge = ({onPress, activeCategory, item}: Props) => {
  const isBadgeActive = activeCategory === item;
  return (
    <TouchableWithoutFeedback onPress={() => onPress(item)}>
      <View
        style={[
          styles.container,
          isBadgeActive && {
            backgroundColor: theme.colors.primary.regular,
          },
        ]}>
        <Text
          style={[styles.text, activeCategory === item && styles.textActive]}>
          {item}
        </Text>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    borderRadius: theme.radius,
    borderWidth: 0.2,
    borderColor: theme.colors.primary.regular,
  },
  text: {
    padding: theme.spacing.sm,
    textTransform: 'uppercase',

    color: theme.colors.primary.regular,
  },
  textActive: {
    color: theme.colors.base.white,
    textDecorationLine: 'none',
  },
});
