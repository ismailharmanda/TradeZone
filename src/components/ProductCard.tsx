import {
  Text,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
  Image,
} from 'react-native';
import {theme} from 'theme';
import {Product} from 'screens/Products/products';

interface Props {
  onPress?: () => void;
  product: Product;
}

const HEIGHT = Dimensions.get('window').height;

export const ProductCard = ({onPress, product}: Props) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Image
        style={styles.image}
        resizeMode="contain"
        source={{
          uri: product.image,
        }}
      />
      <Text numberOfLines={1} style={styles.text}>
        {product.title}
      </Text>
      <Text style={styles.text}>{product.price} TL</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: theme.radius,
    overflow: 'hidden',
    backgroundColor: theme.colors.secondary.transparent,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    width: HEIGHT / 10,
    height: HEIGHT / 7,
    borderWidth: 1,
    borderColor: theme.colors.secondary.regular,
  },
  image: {width: HEIGHT / 10, height: HEIGHT / 10},
  button: {
    width: '100%',
    height: theme.height.button,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: theme.radius,
  },
  primary: {
    backgroundColor: theme.colors.primary.regular,
  },
  'primary-outline': {
    backgroundColor: theme.colors.base.white,
    borderWidth: 1,
    borderColor: theme.colors.primary.regular,
  },
  disabled: {
    backgroundColor: theme.colors.primary.transparent,
  },
  text: {
    fontWeight: 'bold',
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.base.dark,
    width: '100%',
    flexWrap: 'wrap',
  },
});
