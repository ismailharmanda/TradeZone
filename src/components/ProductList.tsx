import {useCallback} from 'react';
import {
  FlatList,
  ViewStyle,
  Text,
  ListRenderItem,
  View,
  StyleSheet,
} from 'react-native';
import {Product} from 'screens/Products/products';
import {theme} from 'theme';

interface Props {
  category: string;
  data: Product[];
  renderItem: ListRenderItem<Product>;
  ListEmptyComponent: JSX.Element;
  contentContainerStyle?: ViewStyle;
  listStyle?: ViewStyle;
  activeCategory?: string;
}

export const ProductsList = ({
  category,
  data,
  renderItem,
  ListEmptyComponent,
  contentContainerStyle,
  listStyle,
  activeCategory,
}: Props) => {
  const keyExtractor = useCallback((item: any) => item.id.toString(), []);

  const isCategoryActive = activeCategory === category;

  const ItemSeperator = useCallback(
    () => <View style={styles.itemSeperator} />,
    [],
  );

  return (
    <View
      style={[
        styles.container,
        !!activeCategory && activeCategory !== category && styles.hide,
      ]}>
      <Text style={styles.text}>{category}</Text>
      <FlatList
        ItemSeparatorComponent={ItemSeperator}
        horizontal={!isCategoryActive}
        style={listStyle}
        contentContainerStyle={[
          contentContainerStyle,
          isCategoryActive && styles.activeCategoryContentContainer,
        ]}
        keyExtractor={keyExtractor}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {flexShrink: 1},
  activeCategoryContentContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    gap: theme.spacing.md,
    justifyContent: 'space-between',
  },
  text: {
    fontWeight: 'bold',
    fontSize: theme.text.size.md.fontSize,
    lineHeight: theme.text.size.md.lineHeight,
    color: theme.colors.tertiary.regular,
    textTransform: 'uppercase',
    marginBottom: theme.spacing.md,
  },
  itemSeperator: {
    width: theme.spacing.md,
  },
  hide: {
    display: 'none',
  },
});
