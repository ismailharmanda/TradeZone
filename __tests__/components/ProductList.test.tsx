import React from 'react';
import {render} from 'react-native-testing-library';
import {ProductsList} from 'components/ProductList';
import {View} from 'react-native';
import {theme} from 'theme';

describe('ProductsList', () => {
  const ListEmptyComponent = <View testID="emptyComponent" />;
  const renderItem = jest.fn();

  const category = 'Category 1';
  const data = [
    {
      id: 1,
      title: 'Product 1',
      price: 10,
      image: 'https://example.com/image.png',
      category: 'Category 1',
      description: 'Description 1',
      rating: {rate: 4.5, count: 10},
    },
  ];
  const dataWithMultipleProducts = [
    {
      id: 1,
      title: 'Product 1',
      price: 10,
      image: 'https://example.com/image.png',
      category: 'Category 1',
      description: 'Description 1',
      rating: {rate: 4.5, count: 10},
    },
    {
      id: 2,
      title: 'Product 2',
      price: 10,
      image: 'https://example.com/image2.png',
      category: 'Category 2',
      description: 'Description 2',
      rating: {rate: 4.5, count: 10},
    },
  ];

  it('renders the component correctly', () => {
    const {toJSON} = render(
      <ProductsList
        category={category}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders the ProductsList component', () => {
    const {getByTestId} = render(
      <ProductsList
        category={category}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />,
    );

    expect(getByTestId('productsList')).toBeTruthy();
  });

  it('renders the FlatList component', () => {
    const {getByTestId} = render(
      <ProductsList
        category={category}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />,
    );

    expect(getByTestId('flatList')).toBeTruthy();
  });

  it('renders the category text', () => {
    const {getByText} = render(
      <ProductsList
        category={category}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />,
    );

    expect(getByText(category)).toBeTruthy();
  });

  it('renders the item separator', () => {
    const {getAllByTestId} = render(
      <ProductsList
        category={category}
        data={dataWithMultipleProducts}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />,
    );

    const itemSeparators = getAllByTestId('itemSeparator');
    expect(itemSeparators.length).toBe(dataWithMultipleProducts.length - 1);
  });

  it('extracts the correct keys from items', () => {
    const {getByTestId} = render(
      <ProductsList
        category={category}
        data={dataWithMultipleProducts}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />,
    );

    const flatList = getByTestId('flatList');
    expect(flatList.props.keyExtractor(dataWithMultipleProducts[0])).toBe(
      dataWithMultipleProducts[0].id.toString(),
    );
    expect(flatList.props.keyExtractor(dataWithMultipleProducts[1])).toBe(
      dataWithMultipleProducts[1].id.toString(),
    );
  });

  it('applies active category content container styles', () => {
    const style = {borderWidth: 1};
    const {getByTestId} = render(
      <ProductsList
        contentContainerStyle={style}
        category={category}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
        activeCategory={category}
      />,
    );

    const contentContainer = getByTestId('flatList');

    expect(contentContainer).toHaveProperty('props.contentContainerStyle', [
      style,
      {
        flexDirection: 'row',
        flexWrap: 'wrap',
        gap: 16,
        justifyContent: 'space-between',
      },
    ]);
  });

  it('does not apply active category content container styles when not active', () => {
    const {getByTestId} = render(
      <ProductsList
        category={category}
        data={data}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
        activeCategory="Category 2"
      />,
    );

    const contentContainer = getByTestId('flatList');

    expect(contentContainer).not.toHaveProperty('props.style', {
      flexDirection: 'row',
      flexWrap: 'wrap',
      gap: theme.spacing.md,
      justifyContent: 'space-between',
    });
  });

  it('renders the item separator with the correct styles', () => {
    const {getByTestId} = render(
      <ProductsList
        category={category}
        data={dataWithMultipleProducts}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />,
    );

    const itemSeparator = getByTestId('itemSeparator');

    expect(itemSeparator).toHaveProperty('props.style', {
      width: theme.spacing.md,
    });
  });

  it('renders the ListEmptyComponent when data is empty', () => {
    const {getByTestId} = render(
      <ProductsList
        category={category}
        data={[]}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />,
    );

    expect(getByTestId('emptyComponent')).toBeTruthy();
  });
});
