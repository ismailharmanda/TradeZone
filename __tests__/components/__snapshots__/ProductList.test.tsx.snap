// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList renders the component correctly 1`] = `
<View
  style={
    [
      {
        "flexShrink": 1,
      },
      false,
      undefined,
    ]
  }
  testID="productsList"
>
  <Text
    style={
      {
        "color": "rgb(255,119,5)",
        "fontSize": 16,
        "fontWeight": "bold",
        "lineHeight": 21,
        "marginBottom": 16,
        "textTransform": "uppercase",
      }
    }
  >
    Category 1
  </Text>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={
      <View
        testID="emptyComponent"
      />
    }
    contentContainerStyle={
      [
        undefined,
        false,
      ]
    }
    data={
      [
        {
          "category": "Category 1",
          "description": "Description 1",
          "id": 1,
          "image": "https://example.com/image.png",
          "price": 10,
          "rating": {
            "count": 10,
            "rate": 4.5,
          },
          "title": "Product 1",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    testID="flatList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
